api/geyser/src/main/java/org/geysermc/geyser/api/item/custom/FullyCustomItemData.java

```diff
@@ -0,0 +1,198 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.api.item.custom;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.geysermc.geyser.api.GeyserApi;
+
+import java.util.OptionalInt;
+import java.util.Set;
+
+/**
+ * This is used to store data for a custom item.
+ */
+public interface FullyCustomItemData extends CustomItemData {
+    /**
+     * Gets the java identifier for this item.
+     *
+     * @return The java identifier for this item.
+     */
+    @NonNull String identifier();
+
+    /**
+     * Gets the java item id of the item.
+     *
+     * @return the java item id of the item
+     */
+    int javaId();
+
+    /**
+     * Gets the stack size of the item.
+     *
+     * @return the stack size of the item
+     */
+    int stackSize();
+
+    /**
+     * Gets the max damage of the item.
+     *
+     * @return the max damage of the item
+     */
+    int maxDamage();
+
+    /**
+     * Gets the tool type of the item.
+     *
+     * @return the tool type of the item
+     */
+    @Nullable String toolType();
+
+    /**
+     * Gets the tool tier of the item.
+     *
+     * @return the tool tier of the item
+     */
+    @Nullable String toolTier();
+
+    /**
+     * Gets the armor type of the item.
+     *
+     * @return the armor type of the item
+     */
+    @Nullable String armorType();
+
+    /**
+     * Gets the armor tier of the item.
+     *
+     * @return the armor tier of the item
+     */
+    @Nullable String armorTier();
+
+    /**
+     * Gets the armor protection value of the item.
+     *
+     * @return the armor protection value of the item
+     */
+    int protectionValue();
+
+    /**
+     * Gets the item's translation string.
+     *
+     * @return the item's translation string
+     */
+    @Nullable String translationString();
+
+    /**
+     * Gets the repair materials of the item.
+     *
+     * @return the repair materials of the item
+     */
+    @Nullable Set<String> repairMaterials();
+
+    /**
+     * Gets the item's creative category, or tab id.
+     *
+     * @return the item's creative category
+     */
+    @NonNull OptionalInt creativeCategory();
```

Comment:
Same as stated earlier - `Optional` is not something we're using in the API.

Reply:
How do we make it not set then? 0 is a creative category id.

Reply:
Could use a nullable `Integer`.

Reply:
Right. Ok